src/swap/components/SwapButton.tsx

```diff
@@ -30,18 +38,35 @@ export function SwapButton({ disabled = false, onSubmit }: SwapButtonReact) {
     }
   }, [address, fromAmount, fromToken, setError, onSubmit, toToken]);
 
+  const isDisabled =
+    !fromAmount ||
+    !fromToken ||
+    !toAmount ||
+    !toToken ||
+    disabled ||
+    swapQuoteLoadingState?.isFromQuoteLoading ||
+    swapQuoteLoadingState?.isToQuoteLoading;
+
+  const isLoading =
+    swapQuoteLoadingState?.isFromQuoteLoading ||
+    swapQuoteLoadingState?.isToQuoteLoading;
```

Comment:
we want loading for `onSubmit` and not on `getSwapQuote`

Reply:
oh woops lol

Reply:
fixed !